### PR TITLE
Rspec for Doctor 

### LINKED ISSUE #7: [1pt] Rspec for Doctor (Melaku)
- model test for Doctor model.
- Requests test for Doctor controller.
- Add Factory-bot-rails to gemfiles.
- Add a faker gem to gemfiles.
- Create factories.rb files.

### PR DESCRIPTION
**Pull request by Melaku05**
- model test for Doctor model.
- Requests test for Doctor controller.
- Add Factory-bot-rails to gemfiles.
- Add a faker gem to gemfiles.
- Create factories.rb  files.

### CODE REVIEW

File: spec/requests/doctors_spec.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        username: 'John Doe',
+        email: 'test@email.com',
+        password: 'Password1'
+      }
+    }.to_json, headers: {
+      'Content-Type': 'application/json',
+      Accept: 'application/json'
+    }
+  end
+
```
Comment (aboongm) [inline spec/requests/doctors_spec.rb:16]: Nice implementing the the user to log in

File: spec/requests/doctors_spec.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  describe 'GET/doctors' do
+    it 'should return a list of doctors' do
+      FactoryBot.create(:doctor, name: 'Dr. John Doe', detail: 'Dentist',
+                                 photo: 'https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png', city: 'New York', specialization: 'Dentist', fee: '100')
+      FactoryBot.create(:doctor, name: 'Dr. Jane Doe', detail: 'Dentist',
+                                 photo: 'https://www.google.com/images/branding/googlelogo/2x/googlelogo_color_272x92dp.png', city: 'New York', specialization: 'Dentist', fee: '100')
+      get '/doctors'
+      expect(response).to have_http_status(200)
```
Comment (aboongm) [inline spec/requests/doctors_spec.rb:24]: Great using the Factorybot

**Review by aboongm — APPROVED** [review]
Hi Teammates,  👏 👏 

The milestone is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

request_changes_1

## Highlights

## Optional suggestions


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/doctors_spec.rb (hunk 2 above)
Comment (Melaku05) [reply spec/requests/doctors_spec.rb:24]: Thank you, brother.

File: spec/requests/doctors_spec.rb (hunk 1 above)
Comment (Melaku05) [reply spec/requests/doctors_spec.rb:16]: Thank you, for four feedback.

**Pull request merged**